src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        linkToTheEnd(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToAdd(index);
+        if (size == index) {
+            linkToTheEnd(value);
+        } else {
+            linkToTheMiddle(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        addAllByIndex(size, list);
     }
```

Comment:
You don`t need helpful method, just reuse simple method add() in for each loop

Reply:
OK. So in this case I actually don't need this method (because I add list at the end of my linkedList). **But if I want to insert my list to devine position in linked list I need to use variant that I wrote below, right?** 

_(becouse to add it in the middle I need to use add(int index) that have loop inside, and I also have loop in addAll, so it's two loops - O(n^2) instead of one loop - O(n) in variant below)_